utils.py

```diff
@@ -453,3 +454,17 @@ def get_all_language_codes_and_names():
 def unescape_encoded_uri_component(escaped_string):
     """Unescape a string that is encoded with encodeURIComponent."""
     return urllib.unquote(escaped_string).decode('utf-8')
+
+
+cache_slug = None  # pylint: disable=invalid-name
+def get_cache_slug():
```

Comment:
You know, separate but related to this discussion, I was wondering: should we use GAE's memcache to cache the slug instead? I feel like that might be better, and then your tests here would be cleaner. We have a memcache service in core/platform that you could use.